In [36]:
import pandas as pd
import numpy as np

df = pd.read_csv('titanic.csv')
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

In [37]:
df[df['Embarked'] == 'C'].groupby('Sex')['Age'].mean()

Sex
female    28.344262
male      32.998841
Name: Age, dtype: float64

In [38]:
df[df['Embarked'] == 'C']['Age'].mean()

30.81476923076923

In [39]:
(28.344262 + 32.998841)/2

30.6715515

# Question: Why we got 30.81 for average age of passengers?

hint: determine how many non Nan values exists for the age of female passengers in C

hint: determine how many not Nan values exists for the age of male passengers in C

In [71]:
df[(df['Embarked'] == 'C') & (df['Sex'] == 'female')].describe()

PassengerId   Survived     Pclass        Age      SibSp      Parch  \
count    73.000000  73.000000  73.000000  61.000000  73.000000  73.000000   
mean    437.315068   0.876712   1.726027  28.344262   0.547945   0.493151   
std     257.119021   0.331042   0.916822  15.770830   0.624524   0.766001   
min       2.000000   0.000000   1.000000   0.750000   0.000000   0.000000   
25%     257.000000   1.000000   1.000000  17.000000   0.000000   0.000000   
50%     381.000000   1.000000   1.000000  24.000000   0.000000   0.000000   
75%     642.000000   1.000000   3.000000  40.000000   1.000000   1.000000   
max     880.000000   1.000000   3.000000  60.000000   2.000000   3.000000   

             Fare  
count   73.000000  
mean    75.169805  
std     83.574380  
min      7.225000  
25%     15.245800  
50%     56.929200  
75%     91.079200  
max    512.329200

In [54]:
df[(df['Embarked'] == 'C') & (df['Sex'] == 'male')].describe()

AttributeError: 'DataFrame' object has no attribute 'view_counts'

In [70]:
df[(df['Embarked'] == 'C') & (df['Sex'] == 'female')]['Age'].value_counts()

24.00    5
30.00    3
18.00    3
17.00    3
39.00    2
23.00    2
0.75     2
54.00    2
44.00    2
38.00    2
50.00    2
15.00    2
14.00    2
22.00    2
14.50    1
49.00    1
58.00    1
3.00     1
31.00    1
32.00    1
29.00    1
35.00    1
19.00    1
56.00    1
40.00    1
48.00    1
27.00    1
9.00     1
13.00    1
21.00    1
4.00     1
52.00    1
5.00     1
36.00    1
1.00     1
42.00    1
60.00    1
45.00    1
28.00    1
16.00    1
41.00    1
Name: Age, dtype: int64